pkg/util/uri.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import "strings"
+
+// IsValidIPv6AddressInURI reports whether hostPort is a valid IPv6 address in URI.
+// See: https://www.ietf.org/rfc/rfc2732.txt.
+func IsValidIPv6AddressInURI(hostPort string) bool {
```

Comment:
What about [`net.ParseIP`](https://pkg.go.dev/net#ParseIP)?

Reply:
The main purpose of this function is to determine the format in the URI, not whether it is an IP.

Reply:
Added in IsIPv6Address.